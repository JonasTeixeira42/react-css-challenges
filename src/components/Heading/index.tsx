import React from 'react';

import * as S from './styles';

const Heading = ({ title = 'Hello World' }) => {
  return <S.Wrapper>{title}</S.Wrapper>;
};

export default Heading;
