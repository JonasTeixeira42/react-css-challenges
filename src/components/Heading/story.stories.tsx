import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import Heading from '.';

export default {
  title: 'Heading',
  component: Heading,
  args: {
    title: 'React Boilerplate',
  },
  argTypes: {
    title: {
      type: {
        name: 'string',
      },
    },
  },
} as Meta;

export const Default: Story = (args) => <Heading {...args} />;
