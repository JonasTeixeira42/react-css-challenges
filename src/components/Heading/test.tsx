import React from 'react';
import { render, screen } from '@testing-library/react';

import Heading from '.';

describe('<Heading />', () => {
  it('should render the heading', () => {
    const { container } = render(<Heading />);

    expect(
      screen.getByRole('heading', { name: /hello world/i }),
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', { name: /hello world/i }),
    ).toHaveStyleRule('color', '#333');

    expect(container.firstChild).toMatchSnapshot();
  });
});
