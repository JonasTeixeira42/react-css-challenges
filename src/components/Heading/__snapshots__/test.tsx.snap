// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render the heading 1`] = `
.c0 {
  color: #333;
}

<h1
  class="c0"
>
  Hello World
</h1>
`;
